check_install.py

```diff
@@ -24,7 +24,18 @@ def main():
     cmd_prompt = "%s@sonic:~\$ $" % args.u
     grub_selection = "The highlighted entry will be executed"
 
-    p = pexpect.spawn("telnet 127.0.0.1 %s" % args.p, timeout=600, logfile=sys.stdout)
+    i = 0
+    while i < 10:
+        try:
+            p = pexpect.spawn("telnet 127.0.0.1 %s" % args.p, timeout=600, logfile=sys.stdout)
+            break
+        except Exception as e:
+            print str(e)
+            time.sleep(1)
+            i += 1
+
+    if i == 10:
+        raise
 
```

Comment:
Not really matters too much. You could do
while True:
   try
     ...
   except
     ...
     i += 1
     if i == 10:
        raise

Up to you if you want to change. Not a big deal for me.

Reply:
yes, your suggestion is easier to read.